### PR TITLE
feat: OpenSearch vector db support

### PR DESCRIPTION
**Pull request by dtaivpp**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [X] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [X] **Description:** Provide a concise description of the changes made in this pull request.
- [X] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [ ] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources? - Not yet, I wanted to wait for it to be merged.
- [X] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [ ] **Testing:** Have you written and run sufficient tests for validating the changes?
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [X] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **feat**: Introduces a new feature or enhancement to the codebase

# Changelog Entry

### Description

- Adding support for OpenSearch as a RAG data store

### Added

- OpenSearch data store connector, relevant configuration, OpenSearch-py dependency

---
- [Attach any relevant screenshots or videos demonstrating the changes]

### CODE REVIEW

**Comment by dtaivpp** [comment]
@tjbck Rebased on Dev

File: backend/requirements.txt
Comment (tjbck) [inline backend/requirements.txt:1]: Unwanted changes.

Comment (dtaivpp) [reply backend/requirements.txt:1]: Good catch @tjbck botched the merge. Have those updated now.

File: backend/open_webui/apps/rag/vector/connector.py
Comment (tjbck) [inline backend/open_webui/apps/rag/vector/connector.py:1]: What is this file for? `/rag` does not exist.

File: pyproject.toml [hunk 1]
```diff
@@ -49,6 +49,7 @@ dependencies = [
     "fake-useragent==1.5.1",
     "chromadb==0.5.9",
     "pymilvus==2.4.7",
+    "opensearch-py==2.7.1"
```
Comment (tjbck) [inline pyproject.toml:52]: Missing comma here.

**Comment by tjbck** [comment]
I think once the comments are resolved, we're good to go! Thanks!

**Comment by doxav** [comment]
It seems like this PR support username/password authentication but not OAuth2, trusted header proxies, and bearer tokens. It would allow to control data access security per user on the OpenSearch side (Opensearch secuirty plugin). Do you plan to do it or would you like a suggestion of implementation ?

File: backend/open_webui/apps/rag/vector/connector.py
Comment (dtaivpp) [reply backend/open_webui/apps/rag/vector/connector.py:1]: 100%, accidentally added that back in. Got that worked out.

File: pyproject.toml (hunk 1 above)
Comment (dtaivpp) [reply pyproject.toml]: ack

**Comment by dtaivpp** [comment]
Hey @doxav! Thanks for the feedback but this is more of a first iteration for OpenSearch. I am mostly doing this on a personal level as I am going to be using OpenSearch locally. If you could use those features feel free to contribute once it's merged 😄 

@tjbck I think I've got this polished off but thank you for your careful review so far!

**Comment by tjbck** [comment]
LGTM, Thanks for the PR!

**Pull request merged**